tests/baselines/reference/docker/vscode.log

```diff
@@ -4,39 +4,39 @@ yarn run vX.X.X
 $ gulp compile --max_old_space_size=4095
 [XX:XX:XX] Node flags detected: --max_old_space_size=4095
 [XX:XX:XX] Using gulpfile /vscode/gulpfile.js
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(222,5): Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(225,7): Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(229,7): Type 'null' is not assignable to type 'string'.
 [XX:XX:XX] Error: /vscode/src/vs/workbench/contrib/search/common/searchModel.ts(609,21): Type 'FileMatch | FileMatch[]' must have a '[Symbol.iterator]()' method that returns an iterator.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/browser/parts/titlebar/titlebarPart.ts(470,5): Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/base/parts/quickinput/browser/quickInputList.ts(474,3): Type '"none" | null' is not assignable to type 'string'.
-  Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/base/browser/ui/splitview/splitview.ts(113,3): Type '"none" | null' is not assignable to type 'string'.
-  Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/editor/browser/controller/pointerHandler.ts(45,45): Property 'msTouchAction' does not exist on type 'CSSStyleDeclaration'. Did you mean 'touchAction'?
-[XX:XX:XX] Error: /vscode/src/vs/editor/browser/controller/pointerHandler.ts(46,45): Property 'msContentZooming' does not exist on type 'CSSStyleDeclaration'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/workbench.web.api.ts(249,25): Type 'readonly ICommand[] | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.
-[XX:XX:XX] Error: /vscode/src/vs/base/browser/ui/splitview/splitview.ts(113,3): Type '"none" | null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(222,5): Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(225,7): Type 'null' is not assignable to type 'string'.
-[XX:XX:XX] Error: /vscode/src/vs/workbench/electron-browser/parts/titlebar/titlebarPart.ts(229,7): Type 'null' is not assignable to type 'string'.
+[XX:XX:XX] Error: /vscode/src/vs/base/browser/ui/tree/asyncDataTree.ts(215,2): Type '{ collapseByDefault: true; identityProvider: { getId(el: IAsyncDataTreeNode<TInput, T>): { toString(): string; }; } | undefined; dnd: AsyncDataTreeNodeListDragAndDrop<TInput, T> | undefined; ... 31 more ...; horizontalScrolling?: boolean | undefined; } | undefined' is not assignable to type 'IObjectTreeOptions<IAsyncDataTreeNode<TInput, T>, TFilterData> | undefined'.
```

Comment:
Parse and emit of `!` following `?.` is now different, meaning that an expression now includes `undefined` in its type and requires additional code to get rid of it.